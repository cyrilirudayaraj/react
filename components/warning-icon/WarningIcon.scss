@import '~@athena/forge/sass/forge-abstracts';
.warning-icon {
  display: inline-block;
  .fe_c_icon {
    .fe_c_icon__icon {
      fill: #ffab00;
    }
    .fe_c_icon__icon--invert {
      fill: #fff;
    }
  }
}
