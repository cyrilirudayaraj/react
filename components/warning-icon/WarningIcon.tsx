import React, { Fragment } from 'react';
import { Tooltip, Icon } from '@athena/forge';
import './WarningIcon.scss';

const WarningIcon = (props: any): JSX.Element => {
  return (
    <div className="warning-icon">
      <Fragment>
        {props.warning && (
          <Tooltip
            text={props.tooltip}
            id={props.label}
            className="requirement-tip"
          >
            <Icon
              icon="Critical"
              className="requirement-icon"
              height={props.height}
              width={props.width}
            />
          </Tooltip>
        )}
      </Fragment>
    </div>
  );
};

export default WarningIcon;
