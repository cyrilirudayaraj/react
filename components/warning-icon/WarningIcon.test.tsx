import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import WarningIcon from './WarningIcon';

configure({ adapter: new Adapter() });
let wrapper: any;

describe('snapshot test <WFInput>', () => {
  it('should render as expected', () => {
    wrapper = shallow(
      <WarningIcon
        warning={true}
        label={'Task ID'}
        tooltip={'Escalated P0 Task'}
        height={22}
        width={25}
      />
    );
    expect(wrapper.debug()).toMatchSnapshot();
  });
});
