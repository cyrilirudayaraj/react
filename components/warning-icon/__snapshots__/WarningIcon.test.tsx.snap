// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test <WFInput> should render as expected 1`] = `
"<div className=\\"warning-icon\\">
  <ThemeConsumerHOC text=\\"Escalated P0 Task\\" id=\\"Task ID\\" className=\\"requirement-tip\\">
    <Icon icon=\\"Critical\\" className=\\"requirement-icon\\" height={22} width={25} iconSet={{...}} />
  </ThemeConsumerHOC>
</div>"
`;
